.github/workflows/cd.yaml

```diff
@@ -38,14 +37,98 @@ jobs:
       - uses: actions/checkout@v1
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set current date as env variable $TODAY
+        run: echo "TODAY=$(date +'%y-%m-%d')" >> $GITHUB_ENV
       - env:
+          CLUSTER: zq-test
+          NAMESPACE: ci-${{ github.run_number }}
           ZQD_ECR_HOST: ${{ secrets.AWS_ECR_HOST }}
-          ZQD_DATA_URI: s3://zqd-demo-1/mark/zqd-meta
-          ZQD_K8S_USER: ci-master
-          ZQD_TEST_CLUSTER: zq-test.us-east-2.eksctl.io
+          ZQD_DATA_URI: s3://zq-ci-master/${{ env.TODAY }}.${{ github.run_number }}
         run: |
-          aws eks --region us-east-2 update-kubeconfig --name zq-test
-          kubectl config set-context --current --namespace=ci-master
+          set -x
+          aws eks --region us-east-2 update-kubeconfig --name $CLUSTER
+          kubectl create ns $NAMESPACE || :
+          kubectl config set-context --current --namespace=$NAMESPACE
+          make helm-uninstall
+          make helm-install-recruiter
+          make helm-install-root
+          make helm-install-worker
+  eks-test-zapi-1:
+    # Run smoke tests for zqd image prior to deploying to ci-master.
+    needs: eks-test-deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.15'
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set current date as env variable
+        run: echo "TODAY=$(date +'%y-%m-%d')" >> $GITHUB_ENV
+      - env:
+          CLUSTER: zq-test
+          NAMESPACE: ci-${{ github.run_number }}
+          SOURCE_LOG_1: s3://brim-sampledata/wrccdc/zeek-logs/files.log.gz
+          SPACE: files
+          ZQD_DATA_URI: s3://zq-ci-master/${{ env.TODAY }}.${{ github.run_number }}
+        run: |
+          set -x
+          make install
+          aws eks --region us-east-2 update-kubeconfig --name $CLUSTER
+          kubectl config set-context --current --namespace=$NAMESPACE
+          kubectl scale --replicas=4 deployment/worker-zqd
+          kubectl port-forward svc/recruiter-zqd 8020:9867 &
+          kubectl port-forward svc/root-zqd 9867:9867 &
+          sleep 2 # wait for port-forwards to complete
+          zapi ls
+          zapi new -k archivestore -d $ZQD_DATA_URI $SPACE
+          # Post an archive of about 1.6 GB to run test queries.
+          zapi -s $SPACE post $SOURCE_LOG_1
+          # There should be 4 workers running now.
+          [[ 4 == $(curl http://localhost:8020/recruiter/listfree | jq -r ".workers" | jq length) ]]     
```

Comment:
The next 5 commands are the smoke test. A basic search, using different numbers of workers, followed by a count() and a group-by query. we are not trying for extensive test coverage here: we just want to fail early if the basic distributed functionality is not working on K8s. (Similar queries are also run locally with ZTest.)

Reply:
I filed #1934 to track follow up work to convert these to tagged ztests.